File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/order/repository/condition/FindMyPageOrderCondition.java
package band.gosrock.domain.domains.order.repository.condition;


import lombok.Builder;
import lombok.Getter;

@Getter
public class FindMyPageOrderCondition {
    private final Long userId;
    //
    private final Boolean showing;

    @Builder
    public FindMyPageOrderCondition(Long userId, Boolean showing) {
        this.userId = userId;
        this.showing = showing;
    }

    public static FindMyPageOrderCondition onShowing(Long userId) {
        return FindMyPageOrderCondition.builder().showing(Boolean.TRUE).userId(userId).build();
    }

    public static FindMyPageOrderCondition notShowing(Long userId) {
        return FindMyPageOrderCondition.builder().showing(Boolean.FALSE).userId(userId).build();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/order/repository/condition/FindMyPageOrderCondition.kt
package band.gosrock.domain.domains.order.repository.condition

data class FindMyPageOrderCondition(
    val userId: Long,
    val showing: Boolean,
) {
    companion object {
        @JvmStatic
        fun onShowing(userId: Long): FindMyPageOrderCondition =
            FindMyPageOrderCondition(userId = userId, showing = true)

        @JvmStatic
        fun notShowing(userId: Long): FindMyPageOrderCondition =
            FindMyPageOrderCondition(userId = userId, showing = false)
    }
}
